homeassistant/components/nasweb/__init__.py

```diff
@@ -0,0 +1,92 @@
+"""The NASweb integration."""
+from __future__ import annotations
+
+import logging
+
+from webio_api import WebioAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+
+from .const import DOMAIN, MANUFACTURER, NASWEB_CONFIG_URL
+from .coordinator import NASwebCoordinator
+from .helper import (
+    deinitialize_nasweb_data_if_empty,
+    get_integration_webhook_url,
+    initialize_nasweb_data,
+)
+from .nasweb_data import NASwebData
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up NASweb from a config entry."""
+
+    hass.data.setdefault(DOMAIN, NASwebData())
+    nasweb_data: NASwebData = hass.data[DOMAIN]
+    if not nasweb_data.is_initialized():
+        initialize_nasweb_data(hass)
+
+    webio_api = WebioAPI(
+        entry.data[CONF_HOST], entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD]
+    )
+    if not await webio_api.check_connection():
+        raise ConfigEntryNotReady(f"[{entry.data[CONF_HOST]}] Check connection failed")
+    if not await webio_api.refresh_device_info():
+        _LOGGER.error("[%s] Refresh device info failed", entry.data[CONF_HOST])
```

Comment:
When `refresh_device_info` fails after successfull `check_connection` that usually means invalid authentication or some internal issue which user won't be able to fix. Added issue creation with short descriptions of these errors. Invalid authentication currently requires user to delete current entry and creating new one. I was thinking about adding Option Flow within future PRs which would allow user to change host/user/password, to avoid recreating entry when device configuration changes.